crowbar_framework/app/models/api/node.rb

```diff
@@ -110,43 +164,22 @@ def upgrade
         return false
       end
 
-      # wait until the OS upgrade is finished
-      upgrade_failure = false
-      begin
-        Timeout.timeout(600) do
-          loop do
-            out = @node.run_ssh_cmd("test -e /var/lib/crowbar/upgrade/node-upgraded-ok")
-            if out[:exit_code].zero?
-              save_node_state("Package upgrade was successful.")
-              break
-            end
-            out = @node.run_ssh_cmd("test -e /var/lib/crowbar/upgrade/node-upgrade-failed")
-            if out[:exit_code].zero?
-              upgrade_failure = true
-              save_error_state("Installation of node #{@node.name} failed")
-              break
-            end
-            sleep(5)
-          end
-        end
-      rescue Timeout::Error
-        save_error_state("Error during upgrading node. Action did not finish after 10 minutes")
-        return false
-      end
-
-      return false if upgrade_failure
+      return false unless reboot_and_wait
 
-      # reboot the node after successful upgrade
-      ssh_status = @node.ssh_cmd("/sbin/reboot")
-      if ssh_status[0] != 200
-        save_error_state("Failed to reboot the machine. Could not ssh.")
+      unless post_upgrade
```

Comment:
Now I realized this is the wrong time to execute post upgrade actions (aka drbd sync and pacemaker start). Old resources need to be deleted (from the other node) before we start pacemaker...